hadoop.yaml

```diff
@@ -0,0 +1,115 @@
+package:
+  name: hadoop
+  version: 3.4.0
+  epoch: 0
+  description: Open Source Identity and Access Management For Modern Applications and Services
+  copyright:
+    - license: Apache-2.0
+  target-architecture:
+    # note: skipping aarch64 because frontend-maven-plugin doesn't work on aarach arch.
+    - x86_64
+  dependencies:
+    runtime:
+      - bash
+      - curl
+      - openjdk-8-default-jvm
+
+environment:
+  contents:
+    packages:
+      - bash
+      - busybox
+      - ca-certificates-bundle
+      - maven
+      - openjdk-8
+  environment:
+    LANG: en_US.UTF-8
+    JAVA_HOME: /usr/lib/jvm/java-1.8-openjdk
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/hadoop
+      tag: rel/release-${{package.version}}
+      expected-commit: bd8b77f398f626bb7791783192ee7a5dfaeec760
+
+  - uses: maven/pombump
+    with:
+      patch-file: pombump-deps.yaml
+      properties-file: pombump-properties.yaml
+      pom: hadoop-project/pom.xml
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/share/java/hadoop
+      mkdir -p ${{targets.destdir}}/usr/bin
+      mkdir -p ${{targets.destdir}}/var/log/hadoop
+      mvn -Pdist,src,yarn-ui -DskipTests -Dtar -Dmaven.javadoc.skip=true clean package -e -X
+      tar -xf hadoop-dist/target/hadoop-${{package.version}}.tar.gz --strip-components=1 -C ${{targets.destdir}}/usr/share/java/hadoop
+
+subpackages:
+  - name: hadoop-runner
+    description: hadoop runner package
+    dependencies:
+      runtime:
+        - byteman
+        - byteman-compat
+        - coreutils
+        - py3-robotframework
+        - dumb-init
+        - python3
+        - py3-pip
+        - krb5
+        - sudo
+        - netcat-openbsd
+        - jq
+        - async-profiler
+    pipeline:
+      - uses: git-checkout
+        with:
+          repository: https://github.com/apache/hadoop
+          # this branch is where the hadoop-runner base image comes from.
+          # for update please look into the PR that are updating the base image.
+          # previous work: https://github.com/apache/hadoop/pull/6910
+          branch: docker-hadoop-runner-jdk11-u2204
+          expected-commit: 5bb6f450340fa3deb931571d08d42a3f358af8bf
```

Comment:
given this is a branch where upstream supports docker related configuration. I didn't find a way to poll based on branches. For now, I think this is stable because they are making new changes on another branch but if there's a way then please let me know about it. I'll implement the same. The central point here is that we need to poll on branches that start with `docker-` and then checkout that and build it.
From my attempt of looking into it, I find that doing it manually is the way to go for now because they also remove something in new branches which we depend on. 
For example: This (ref: https://github.com/apache/hadoop/tree/docker-hadoop-3.4) is latest `docker-` branch but python configuration on which we depend is missing from this branch.

Reply:
I'm not sure I follow. We'd create a separate package and then just poll like this:

```yaml
update:
  enabled: true
  schedule: daily
  git: {}
```

We shouldn't need to pass anything else as it will leverage whatever we pass to `git-checkout` for polling

Reply:
thank you for the review. New package is tracked [here](https://github.com/wolfi-dev/os/pull/30294) and I've included the same as runtieme dependency to this package (https://github.com/wolfi-dev/os/pull/30294)